src/core/ui/components/drop-down/index.tsx

```diff
@@ -0,0 +1,353 @@
+import { useEffect, useMemo, useRef, useState } from 'react';
+import { ScaleLoader } from 'react-spinners';
+import { useCoreSlashAuth } from '../../context/core-slashauth';
+import { shortenEthAddress } from '../../../../shared/utils/eth';
+import { profilePicturePlaceholder } from './profilePicturePlaceholder';
+import { chevronDown } from './chevronDown';
+import { googleIcon } from './googleIcon';
+import { twitterIcon } from './twitterIcon';
+import { copyIcon } from './copyIcon';
+import { plusIcon } from './plusIcon';
+import { gearIcon } from './gearIcon';
+import { logoutIcon } from './logoutIcon';
+
+type TestUser = {
+  name?: string;
+  loggedIn: boolean;
+  wallet?: { default: string };
+  email?: string;
+  social?: {
+    google: string;
+    twitter: string;
+  };
+};
+export const DropDown = () => {
+  const user: TestUser = testUser.walletOnly;
+  const [isOpen, setIsOpen] = useState(false);
+  const context = useCoreSlashAuth();
+  const { isReady, logout, openSignIn, connectWallet } = context;
+  const [wallet, walletDisplay] = useMemo(() => {
+    if (user.wallet) {
+      const address: string = user.wallet.default.split(':')[1]; //undefined possible
```

Comment:
if undefined is possible, let's protect against it!

Reply:
updated